gcc/rust/rust-session-manager.cc

```diff
@@ -883,7 +884,11 @@ Session::expansion (AST::Crate &crate)
       auto ctx = Resolver2_0::NameResolutionContext ();
 
       if (flag_name_resolution_2_0)
-	Resolver2_0::Early (ctx).go (crate);
+	{
+	  Resolver2_0::Early early (ctx);
+	  early.go (crate);
+	  macro_errors = early.get_macro_resolve_errors ();
```

Comment:
we need to clear out the macro resolve errors each time we run through the fixed point, otherwise we are going to emit errors which would have been resolved in a later run, aren't we?

Reply:
Mmmh no? From what I remember I tested this edge case. I may be wrong though.

`macro_errors` get cleared by the assignation and the internal vector from `get_macro_resolve_errors` is cleared anyway because of the new instance of the early resolver in the loop. If we change this instantiation to put it out of the loop then yes we should provide a function to clear the vector.